src/snowflake/snowpark/context.py

```diff
@@ -42,33 +42,47 @@
 
 def configure_development_features(
     *,
-    enable_dataframe_trace_on_error: bool = True,
-    enable_eager_schema_validation: bool = True,
-    enable_trace_sql_errors_to_dataframe: bool = True,
+    enable_eager_schema_validation: bool = False,
+    enable_dataframe_trace_on_error: bool = False,
+    enable_trace_sql_errors_to_dataframe: bool = False,
 ) -> None:
     """
     Configure development features for the session.
 
     Args:
-        enable_dataframe_trace_on_error: If True, upon failure, we will add most recent dataframe
-            operations to the error trace. This requires AST collection to be enabled in the
-            session which can be done using `session.ast_enabled = True`.
         enable_eager_schema_validation: If True, dataframe schemas are eagerly validated by querying
             for column metadata after every dataframe operation. This adds additional query overhead.
+        enable_dataframe_trace_on_error: If True, upon failure, we will add most recent dataframe
+            operations to the error trace. This enables the AST collection in the session.
         enable_trace_sql_errors_to_dataframe: If True, we will enable tracing sql compilation errors
-            to the associated dataframe operations.
+            to the associated dataframe operations. This enables the AST collection in the session.
     Note:
         This feature is experimental since 1.33.0. Do not use it in production.
     """
     _logger.warning(
         "configure_development_features() is experimental since 1.33.0. Do not use it in production.",
     )
-    global _enable_dataframe_trace_on_error, _enable_trace_sql_errors_to_dataframe
     global _debug_eager_schema_validation
-    _enable_dataframe_trace_on_error = enable_dataframe_trace_on_error
+    global _enable_dataframe_trace_on_error
+    global _enable_trace_sql_errors_to_dataframe
     _debug_eager_schema_validation = enable_eager_schema_validation
+    _enable_dataframe_trace_on_error = enable_dataframe_trace_on_error
     _enable_trace_sql_errors_to_dataframe = enable_trace_sql_errors_to_dataframe
 
+    if _enable_dataframe_trace_on_error or _enable_trace_sql_errors_to_dataframe:
+        try:
+            session = get_active_session()
+            if session is None:
+                _logger.warning(
+                    "No active session found. Please create a session first to enable development features.",
+                )
+                return
```

Comment:
if there is no active session and I did `configure_development_features(enable_trace_sql_errors_to_dataframe =True)`, and later on I create session, would `enable_trace_sql_errors_to_dataframe` work? or do I need to call this `configure_development_features` again to enable it